tallycash/extension
# Gas price bids on Arbitrum

Resolves https://github.com/tallycash/extension/issues/2367
Resolves https://github.com/tallycash/extension/issues/2373
Resolves https://github.com/tallycash/extension/issues/2428

### Background
Arbitrum has its own unique gas rules so we can't reuse EIP-1559 compatible gas settings implementation. This PR is a refactor of our old code, so it is aligned with what Arbitrum transactions expect.

One problem that was not solved here - I could not find a good price oracle for medium and high gas price bids. The one in the first commit has too low rate limit for us. I would like to avoid looking for another paid provider. The manual calculation follows the rules I see in Debank's price oracle.

### What 

This PR fixes problems with the Arbitrum gas selector.
- Allow changing the gas price bid for Arbitrum transactions
- Gas price is calculated manually as we don't have a good price oracle for now. 
- Reuse legacy network settings component

![image](https://user-images.githubusercontent.com/20949277/199247330-21d5655d-7c76-4130-8796-5746a4ef6fc2.png)

### Testing
- setup transaction on Arbitrum (send ETH), check the gas estimate, it should be close to the estimate on [L2Fees](https://l2fees.info/)
- change network settings and select a higher gas option
- send the transaction and check if the gas bid was the one you selected and if it was higher than the price you paid for gas
![image](https://user-images.githubusercontent.com/20949277/199245170-671c5e4a-298d-4615-9173-b9485996e788.png)

 

Latest build: [extension-builds-2532](https://github.com/tallycash/extension/suites/9073161831/artifacts/419958519) (as of Tue, 01 Nov 2022 17:58:31 GMT).
